examples/face_classification.py

```diff
@@ -0,0 +1,161 @@
+             Face(15, 11.5, 18, 7.5), Face(15, 11.5, 9, 7.5), Face(19.5, 21.5, 9, 16.5), Face(23.5, 11.5, 9, 7.5),  # 12
+             Face(15, 21.5, 13.5, 12), Face(19.5, 16.5, 9, 16.5), Face(19.5, 16.5, 9, 12), Face(15, 16.5, 13.5, 16.5),  # 16
+             Face(23.5, 11.5, 9, 16.5), Face(15, 11.5, 13.5, 12), Face(15, 21.5, 18, 16.5), Face(19.5, 11.5, 18, 16.5),  # 20
+             Face(19.5, 21.5, 9, 7.5), Face(15, 21.5, 18, 12), Face(19.5, 11.5, 13.5, 16.5), Face(15, 16.5, 13.5, 12),  # 24
+             Face(15, 16.5, 18, 16.5), Face(19.5, 21.5, 9, 16.5), Face(15, 16.5, 18, 12), Face(19.5, 16.5, 13.5, 7.5),  # 28
+             Face(15, 21.5, 13.5, 16.5), Face(15, 11.5, 13.5, 12), Face(23.5, 11.5, 18, 12), Face(19.5, 11.5, 13.5, 12),  # 32
+             Face(22.7, 16.5, 16.2, 12), Face(15.9, 12.5, 12.6, 11.1)]  # 34
+    CAT_E1 = [1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 1, 2, 2, 2, 2, 1, 2, 2, 1, 2, 2, 1, 2, 2, 2, 2, 2, 2, 2, 1, 1, 1, 2]
+    CAT_E2 = [2, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 2, 2, 1, 1]
+
+    def __init__(self, env, condition, corrects=None):
+        super().__init__()
+        self.display = env.display
+        self.keyboard = env.keyboard
+        self.selected_categories = corrects or Data(len(self.FACES))
+        self.done = False
+        self.condition = condition
+
+    def run(self, time):
+
+        def is_training():
+            return self.phase == 'Training'
+
+        def handle_key(key):
+            if not is_training():
+                self.selected_categories.add(self.face_num, int(key))
+
+        self.keyboard.add_type_fn(handle_key)
+
+        for phase in self.N_BLOCKS.keys():
+            self.phase = phase
+            for block in range(self.N_BLOCKS[phase]):
+                face_nums = [i for i in range(len(self.FACES))] if not is_training() else self.condition.train_stim
+                random.shuffle(face_nums)
+                for num in face_nums:
+                    self.face_num = num
+                    self.trial_start = self.time()
+                    self.display.clear()
+                    face_visual = self.display.add_face(50, 50, self.FACES[num], isa='face')
+                    self.display.set_attend(face_visual)
+                    self.wait(5)
+                    self.display.clear()
+                    if is_training():
+                        category = self.CAT_E1[num] if self.condition.name[0] == '1' else self.CAT_E2[num]
+                        category_visual = self.display.add_text(50, 50, category)
+                        self.display.set_attend(category_visual)
+                    else:
+                        self.display.add_text(50, 50, '')
+
+
+class FaceClassificationAgent(Agent):
+    MAX_MIN = [(23.5, 15), (21.5, 11.5), (18, 9), (16.5, 7.5)]  # TODO move somewhere? calculate instead of hard code?
```

Comment:
I think it's fine to leave here but I would try to calculate it if the agent has access to that data before the experiment just to eliminate the possibility of error. If not then it's fine to leave I think